File: ParamFileGenerator/Parameters/IBasicParams.cs
using System.Collections.Generic;
using ParamFileGenerator.Modifications;

namespace ParamFileGenerator.Parameters
{
    public interface IBasicParams
    {
        public enum MassTypeList
        {
            Average = 0,
            Monoisotopic = 1
        }

        Params.ParamFileTypes FileType { get; }
        int DMS_ID { get; set; }
        string FileName { get; set; }
        string Description { get; set; }
        EnzymeDetails SelectedEnzymeDetails { get; set; }
        int SelectedEnzymeIndex { get; set; }
        int SelectedEnzymeCleavagePosition { get; set; }
        int MaximumNumberMissedCleavages { get; set; }
        MassTypeList ParentMassType { get; set; }
        MassTypeList FragmentMassType { get; set; }
        DynamicMods DynamicMods { get; set; }
        TermDynamicMods TermDynamicMods { get; set; }
        StaticMods StaticModificationsList { get; set; }
        IsoMods IsotopicModificationsList { get; set; }
        string PartialSequenceToMatch { get; set; }
        List<EnzymeDetails> EnzymeList { get; set; }
        EnzymeDetails RetrieveEnzymeDetails(int index);
    }
}

File: ParamFileGenerator/Parameters/IBasicParams.vb
Public Interface IBasicParams
    Enum MassTypeList As Integer
        Average = 0
        Monoisotopic = 1
    End Enum

    ReadOnly Property FileType As Params.ParamFileTypes
    Property DMS_ID As Integer
    Property FileName As String
    Property Description As String
    Property SelectedEnzymeDetails As EnzymeDetails
    Property SelectedEnzymeIndex As Integer
    Property SelectedEnzymeCleavagePosition As Integer
    Property MaximumNumberMissedCleavages As Integer
    Property ParentMassType As MassTypeList
    Property FragmentMassType As MassTypeList
    Property DynamicMods As DynamicMods
    Property TermDynamicMods As TermDynamicMods
    Property StaticModificationsList As StaticMods
    Property IsotopicModificationsList As IsoMods
    Property PartialSequenceToMatch As String
    Property EnzymeList As EnzymeCollection
    Function RetrieveEnzymeDetails(index As Integer) As EnzymeDetails
End Interface
